File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/client/messages/ClientMessageUpdateDocument.java
package net.stzups.board.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.server.websocket.protocol.client.ClientMessage;
import net.stzups.board.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageUpdateDocument extends ClientMessage {
    private long id;
    private String name;

    public ClientMessageUpdateDocument(ByteBuf byteBuf) {
        super(ClientMessageType.UPDATE_DOCUMENT);
        id = byteBuf.readLong();
        name = readString(byteBuf);
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/client/messages/ClientMessageUpdateDocument.ts
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";
import ByteBuffer from "../../ByteBuffer.js";
import Document from "../../../Document.js";

export default class ClientMessageUpdateDocument extends ClientMessage {
    id: bigint;
    name: string;

    constructor(document: Document) {
        super();
        this.id = document.id;
        this.name = document.name;
    }

    getType(): ClientMessageType {
        return ClientMessageType.UPDATE_DOCUMENT;
    }

    serialize(byteBuffer: ByteBuffer) {
        super.serialize(byteBuffer);
        byteBuffer.writeBigInt64(this.id);
        byteBuffer.writeString8(this.name);
    }
}